Mathlib/Topology/Instances/CantorSet.lean

```diff
@@ -0,0 +1,92 @@
+/-
+Copyright (c) 2024 Jana Göken. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Artur Szafarczyk, Suraj Krishna M S, Jean-Baptiste Stiegler, Isabelle Dubois,
+Tomáš Jakl, Lorenzo Zanichelli, Alina Yan, Emilie Uthaiwat, Jana Göken,
+Filippo A. E. Nuccio
+-/
+import Mathlib.Topology.Metrizable.Basic
+import Mathlib.Topology.Algebra.GroupWithZero
+import Mathlib.Topology.Instances.Real
+
+/-!
+# Ternary Cantor Set
+
+This file defines the Cantor ternary set and proves a few properties.
+
+## Main Definitions
+
+* `preCantorSet n`: The order `n` pre-Cantor set, defined inductively as the union of the images
+  under the functions `(· / 3)` and `(2 * · / 3)`, with `preCantorSet 0 := Set.Icc 0 1`, i.e.
+  `preCantorSet 0` is the unit interval [0,1].
+* `cantorSet`: The ternary Cantor set, defined as the intersection of all pre-Cantor sets.
+-/
+
+/-- The definition of preCantorSet, as explained above. -/
+def preCantorSet : ℕ → Set ℝ
+  | 0 => Set.Icc 0 1
+  | n + 1 => (· / 3) '' preCantorSet n ∪ (fun x ↦ (2 + x) / 3) '' preCantorSet n
+
+@[simp] lemma preCantorSet_zero : preCantorSet 0 = Set.Icc 0 1 := rfl
+@[simp] lemma preCantorSet_succ (n : ℕ) :
+    preCantorSet (n + 1) = (· / 3) '' preCantorSet n ∪ (fun x ↦ (2 + x) / 3) '' preCantorSet n :=
+  rfl
+
+/-- The Cantor set is the intersection of all pre-Cantor sets. -/
```

Comment:
Also this could be more informative: The Cantor sets is the subset of the unit interval created by iteratively removing the open middle third of each interval. Or something along those lines.

Reply:
This is now addressed.